prog/weaveutil/datapath.go

```diff
@@ -21,7 +30,15 @@ func createDatapath(args []string) error {
 		// status to distinguish it for the weave script.
 		os.Exit(17)
 	}
-	return err
+	if err != nil {
+		return err
+	}
+
+	if err := wnet.SetMTU(dpName, mtu); err != nil {
```

Comment:
Good point. In the bash version we do not remove the datapath if setting MTU fails, so I'm leaving as it is.